packages/zero-client/src/client/zero.ts

```diff
@@ -830,42 +829,43 @@ export class Zero<
     return createLogOptions(options);
   }
 
-  preload(
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    query: Query<S, keyof S['tables'] & string, any>,
-    options?: PreloadOptions | undefined,
-  ) {
-    return query[delegateSymbol](this.#zeroContext).preload(options);
+  preload<
+    TTable extends keyof S['tables'] & string,
+    TReturn extends PullRow<TTable, S>,
+    Q extends WithContext<S, TTable, TReturn, TContext>,
+  >(query: Q, options?: PreloadOptions) {
+    const q = query.withContext(this.context);
+    return q[delegateSymbol](this.#zeroContext).preload(options);
   }
 
-  run<Q>(
-    query: Q,
-    runOptions?: RunOptions | undefined,
-  ): Promise<HumanReadable<QueryReturn<Q>>> {
-    return (
-      (query as AnyQuery)
-        // eslint-disable-next-line no-unexpected-multiline
-        [delegateSymbol](this.#zeroContext)
-        .run(runOptions) as Promise<HumanReadable<QueryReturn<Q>>>
-    );
+  run<
+    TTable extends keyof S['tables'] & string,
+    TReturn extends PullRow<TTable, S>,
+    Q extends WithContext<S, TTable, TReturn, TContext>,
+  >(newQuery: Q, runOptions?: RunOptions): Promise<HumanReadable<TReturn>> {
+    const q = newQuery.withContext(this.context);
+    return q[delegateSymbol](this.#zeroContext).run(runOptions);
+  }
+
+  get context(): TContext {
```

Comment:
`context` or `ctx`

It felt too strange to call this ctx here... which kind of implies we should call it context everywhere